### PR TITLE
Add codestart to extension

### PR DESCRIPTION
**Pull request by mcruzdev**
Fixes: https://github.com/quarkiverse/quarkus-openapi-generator/issues/542

Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [our code style](CONTRIBUTING.md#code-style)
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] Subject`
- [ ] Pull Request contains link to the issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-quarkus2` to backport the original PR to the `quarkus2` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: integration-tests/pom.xml [hunk 1]
```diff
@@ -68,6 +68,16 @@
...
+        <dependency>
+          <groupId>io.quarkus</groupId>
+          <artifactId>quarkus-devtools-testing</artifactId>
+          <scope>test</scope>
+        </dependency>
+        <dependency>
+          <groupId>io.quarkus</groupId>
+          <artifactId>quarkus-junit5</artifactId>
+          <scope>test</scope>
+        </dependency>
```
Comment (mcruzdev) [inline integration-tests/pom.xml:80]:
```suggestion
```

File: runtime/src/main/codestarts/quarkus/openapi-generator-codestart/java/README.tpl.qute.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{#include readme-header /}
+
+If you do not have added the `io.quarkus:quarkus-rest-client-jackson` extension in your project, add it first:
+
+```bash
+quarkus ext add io.quarkus:quarkus-rest-client-jackson
+```
```
Comment (hbelmiro) [inline runtime/src/main/codestarts/quarkus/openapi-generator-codestart/java/README.tpl.qute.md:7]: Maybe we should mention the reactive client as well.

File: runtime/src/main/codestarts/quarkus/openapi-generator-codestart/java/src/main/resources/application.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+quarkus:
```
Comment (hbelmiro) [inline runtime/src/main/codestarts/quarkus/openapi-generator-codestart/java/src/main/resources/application.yml:1]: I'm wondering if we shouldn't use application.properties instead. I'm worried about mixing formats (from other extensions). Doesn't codestarts have a standard?

Comment (mcruzdev) [reply runtime/src/main/codestarts/quarkus/openapi-generator-codestart/java/src/main/resources/application.yml:1]: This is the standard of codestarts, at the end, the CLI will merge all properties into the application.properties file. See here https://quarkus.io/guides/extension-codestart#app-config

File: runtime/src/main/codestarts/quarkus/openapi-generator-codestart/java/README.tpl.qute.md (hunk 2 above)
Comment (mcruzdev) [reply runtime/src/main/codestarts/quarkus/openapi-generator-codestart/java/README.tpl.qute.md]: Solved! @hbelmiro

**Review by hbelmiro — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**